src/System.Device.Ports/System.Device.Ports.SerialPort/SerialPort.Settings.cs

```diff
@@ -0,0 +1,728 @@
+        /// </summary>
+        public Parity Parity
+        {
+            get => _parity;
+            set
+            {
+                if (value < Parity.None || value > Parity.Space)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(Parity), Strings.ArgumentOutOfRange_Enum);
+                }
+
+                if (value == _parity)
+                {
+                    return;
+                }
+
+                _parity = value;
+                SetParity(_parity);
+            }
+        }
+
+        /// <summary>
+        /// Set the communication parity
+        /// </summary>
+        /// <param name="parity">The parity value to set</param>
+        protected internal abstract void SetParity(Parity parity);
+
+        /// <summary>
+        /// The data bits
+        /// </summary>
+        public int DataBits
+        {
+            get => _dataBits;
+            set
+            {
+                // 9 data bit is only supported by toggling the parity bit
+                if (_dataBits < MinDataBits || _dataBits > MaxDataBitsNoParity ||
+                    (_dataBits == MaxDataBitsNoParity && Parity != Parity.None))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(DataBits), Strings.InvalidDataBits);
+                }
+
+                if (value == _dataBits)
+                {
+                    return;
+                }
+
+                _dataBits = value;
+                SetDataBits(_dataBits);
+            }
+        }
+
+        /// <summary>
+        /// Set the communication data bits
+        /// </summary>
+        /// <param name="dataBits">The data bits value to set</param>
+        protected internal abstract void SetDataBits(int dataBits);
+
+        /// <summary>
+        /// The stop bits
+        /// </summary>
+        public StopBits StopBits
+        {
+            get => _stopBits;
+            set
+            {
+                if (value < StopBits.One || value > StopBits.OnePointFive)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(StopBits), Strings.ArgumentOutOfRange_Enum);
+                }
+
+                if (value == _stopBits)
+                {
+                    return;
+                }
+
+                _stopBits = value;
+                SetStopBits(_stopBits);
+            }
+        }
+
+        /// <summary>
+        /// Set the communication stop bits
+        /// </summary>
+        /// <param name="stopBits">The stop bits value to set</param>
+        protected internal abstract void SetStopBits(StopBits stopBits);
+
+        /// <summary>
+        /// Gets or sets the break signal state.
+        /// </summary>
+        public bool BreakState
+        {
+            get => _breakState;
+            set
+            {
+                /*
+                if (value == _breakState)
+                {
+                    return;
+                }
+                */
```

Comment:
can you write a comment why this is commented out?

Reply:
I am going to add this comment:
```
In the old implementation SetBreakState is set every time using platform specific code for this reason, this code is commented but I left it here because the old behavior is still to be confirmed.
```